### PR TITLE
Add test scripts for 3888

### PR DESCRIPTION
**Pull request by ShobhitAd**
ATF Test Scripts to check #[3888](https://github.com/smartdevicelink/sdl_core/issues/3888)

This PR is **ready** for review.

### Summary
Scripts to check resumption of `CreateInteractionChoiceSet` with and without `VR` command

### ATF version
develop

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Defects/8_2/3888/common.lua [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+      result = "SUCCESS"
+    }
+  },
+  without_vr_command = {
+    parameters = {
+      interactionChoiceSetID = 1002,
+      choiceSet = {
+        {
+          choiceID = 1002,
+          menuName ="Choice1002"
```
Comment (jacobkeeler) [inline test_scripts/Defects/8_2/3888/common.lua:72]:
```suggestion
          menuName = "Choice1002"
```

File: test_scripts/Defects/8_2/3888/common.lua [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+
+--[[ Common Variables ]]
+m.testCases = {
+  with_vr_command = {
+    parameters = {
+      interactionChoiceSetID = 1001,
+      choiceSet = {
+        {
+          choiceID = 1001,
+          menuName ="Choice1001",
```
Comment (jacobkeeler) [inline test_scripts/Defects/8_2/3888/common.lua:42]:
```suggestion
          menuName = "Choice1001",
```

File: test_scripts/Defects/8_2/3888/common.lua [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+m.registerAppWOPTU = actions.registerAppWOPTU
+m.activateApp = actions.activateApp
+m.spairs = utils.spairs
+m.connectMobile = actions.mobile.connect
+
+--[[ Local Variables ]]
+local addSubMenuParams = {
+  menuID = 1000,
+  position = 500,
+  menuName ="Sub Menu"
```
Comment (jacobkeeler) [inline test_scripts/Defects/8_2/3888/common.lua:31]:
```suggestion
  menuName = "Sub Menu"
```

File: test_scripts/Defects/8_2/3888/common.lua [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+  actions.getMobileSession():ExpectNotification("OnHashChange")
+  :Do(function(_, data)
+      m.hashID = data.payload.hashID
+    end)
+end
+
+function m.unexpectedDisconnect()
+  actions.mobile.disconnect()
+  actions.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered",
+    { appID = actions.getHMIAppId(), unexpectedDisconnect = true })
```
Comment (jacobkeeler) [inline test_scripts/Defects/8_2/3888/common.lua:116]: Is this a standard function anywhere?

File: test_scripts/Defects/8_2/3888/common.lua [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+end
+
+function m.ignitionOff()
```
Comment (jacobkeeler) [inline test_scripts/Defects/8_2/3888/common.lua:119]: Same here, is this already in `actions.lua` or something similar?

File: test_scripts/Defects/8_2/3888/3888_2_CreateInteractionChoiceSet_resumption_ignition_off.lua [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+--[[ Test ]]
+for k, v in common.spairs(common.testCases) do
+  common.Title("Test case: CreateInteractionChoiceSet " .. k)
+  common.Title("Preconditions")
+  common.Step("Clean environment", common.preconditions)
+  common.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+  common.Step("Register App", common.registerAppWOPTU)
+  common.Step("Activate App", common.activateApp)
+
+  common.Step("App requests CreateInteractionChoiceSet " .. k, common.createInteractionChoiceSet, { v })
```
Comment (jacobkeeler) [inline test_scripts/Defects/8_2/3888/3888_2_CreateInteractionChoiceSet_resumption_ignition_off.lua:31]:
```suggestion
  common.Title("Test")
  common.Step("App requests CreateInteractionChoiceSet " .. k, common.createInteractionChoiceSet, { v })
```
Missing from most of these tests

File: test_scripts/Defects/8_2/3888/common.lua [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+      actions.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+  actions.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+  actions.getMobileSession():ExpectNotification("OnHashChange")
+  :Do(function(_, data)
+      m.hashID = data.payload.hashID
+    end)
+end
+
+function m.additionalResumptionConditions(pTimes, pCmdId)
```
Comment (jacobkeeler) [inline test_scripts/Defects/8_2/3888/common.lua:206]:
```suggestion
function m.addSubMenuResumptionFail(pTimes, pCmdId)
```

**Review by jacobkeeler — CHANGES_REQUESTED** [review]

File: test_scripts/Defects/8_2/3888/common.lua (hunk 4 above)
Comment (ShobhitAd) [reply test_scripts/Defects/8_2/3888/common.lua:116]: Seems that the function is defined differently in various common files so there's no base implementation to use. It is defined  in `commonSmoke.lua` but the commonSmoke implementation deletes the mobileSession while this implementation only does a disconnect

File: test_scripts/Defects/8_2/3888/common.lua [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+      timeoutPrompt = {
+        {
+          text = "Timeout Prompt",
+          type = "TEXT"
+        }
+      }
+    }
+end
+    
+    function m.performInteractionVR(pParams)
```
Comment (jacobkeeler) [inline test_scripts/Defects/8_2/3888/common.lua:238]: Still have inconsistent indentation in this part of the file

**Review by jacobkeeler — CHANGES_REQUESTED** [review]

File: test_scripts/Defects/8_2/3888/common.lua (hunk 7 above)
Comment (jacobkeeler) [reply test_scripts/Defects/8_2/3888/common.lua:206]: This function name was never actually changed, just the usages

**Review by jacobkeeler — APPROVED** [review]

**Pull request merged**